### DISCUSSION TITLE
Using `prop` with member functions

### DISCUSSION THREAD

**Discussion by Centronias** [assertk-org/assertk#369] [Ideas]
Heya, I've been using Assertk 0.22 for a while and when upgrading to 0.24, I found that one of my favorite capabilities is now deprecated.

That capability was `fun <T, P> Assert<T>.prop(callable: KCallable<P>)`. I understand why it was removed, but I'm wondering if it wouldn't be possible to get it back with some trickier implementation.

Specifically
```kotlin
fun <T, R, F> Assert<T>.prop(fn: F): Assert<R> where F : (T) -> R, F : KCallable<R> =
  prop(fn.name, fn)
```
.

It's basically the same as the old Callable declaration except that it's guaranteed at compiletime to have the right number and type of arguments.

I've been able to replace all of my old usages with this, and it seems to work, so I am wondering if there's some problem with this approach.
Maybe multiplatform has an issue with it? I'm JVM only.

**Comment by evant** [comment]
Perfectly fine with the idea if it solves the type safety issue, you could write a pr and verify that it works on multiplatform.
